torch/csrc/jit/codegen/cuda/predicate_compute.h

```diff
@@ -38,7 +39,36 @@ class PredicateCompute {
   // reutrns 1
   static std::vector<kir::Bool*> computePredicates(
       const TensorView* tv,
-      const std::vector<Val*>& indices);
+      const std::vector<Val*>& indices,
+      bool use_rfactor);
+
+  static kir::Bool* getInlinePredicate(
+      Expr* expr,
+      const std::vector<kir::ForLoop*>& loops,
+      kir::Bool* thread_pred);
+};
+
+class TORCH_CUDA_API UnrollPredicate {
+ public:
+  static kir::Bool* get(
+      const std::vector<kir::ForLoop*>& outer_loops,
+      kir::ForLoop* unrolled_loop,
+      const std::unordered_map<IterDomain*, IterDomain*>& p2c_root_map);
+
+ private:
+  UnrollPredicate(
+      std::vector<kir::ForLoop*> outer_loops,
+      kir::ForLoop* unrolled_loop,
+      const std::unordered_map<IterDomain*, IterDomain*>& _p2c_root_map);
+
+  void predicateOn(Expr*);
+
+  void openLoop(kir::ForLoop*);
+
+  std::unordered_map<IterDomain*, kir::Bool*> predicates;
```

Comment:
`private:` to separate private methods from private state

Reply:
They're ordered, that may be the best you get from me.